mariadb-native: fix link error on Ubuntu 13.10

Below errors only occurs on Ubuntu 13.10:

$arch-linux-libtool: link: g++ ... -o .libs/mysqltest_embedded \
    ../../libmysqld/.libs/libmysqld.so -ldl

    ../../libmysqld/.libs/libmysqld.so: undefined reference to `dlopen'
    ../../libmysqld/.libs/libmysqld.so: undefined reference to `dlclose'
    ../../libmysqld/.libs/libmysqld.so: undefined reference to `dlerror'
    ../../libmysqld/.libs/libmysqld.so: undefined reference to `dlsym'

GCC/ld verion on the host:
    gcc (Ubuntu/Linaro 4.8.1-10ubuntu8) 4.8.1
    GNU ld (GNU Binutils for Ubuntu) 2.23.52.20130913

This is a strange behavior on Ub13.10, it fails even '-ldl' in the
link command line. Below fix will append '-ldl' to dependency_libs
in libmysqld.la.

Upstream-Status: Submitted [https://mariadb.atlassian.net/browse/MDEV-5362]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
================================================
diff --git a/libmysqld/Makefile.am b/libmysqld/Makefile.am
index 7a2c92e..eee501e 100644
--- a/libmysqld/Makefile.am
+++ b/libmysqld/Makefile.am
@@ -92,7 +92,7 @@ INC_LIB=	$(top_builddir)/regex/libregex.la \
                 @ndbcluster_libs@ @NDB_SCI_LIBS@ \
 		@mysql_embedded_plugin_libs@ \
 		$(libevent_inc_libs) \
-		$(yassl_inc_libs)
+		$(yassl_inc_libs) @LIBDL@
 
 if HAVE_YASSL
 yassl_inc_libs=	$(top_builddir)/extra/yassl/src/libyassl.la \
